File: leetcode_java/java/com/goinhn/leetcode/_1_100/Sol70.java
package com.goinhn.leetcode._1_100;

/**
 * 70.爬楼梯
 *
 * @author goinhn
 * @date 2021/11/20
 */
public class Sol70 {

    /**
     * 动态规划
     * <p>
     * time:O(n)
     * space:O(1)
     */
    private static class Solution1 {
        public int climbStairs(int n) {
            if (n < 3) {
                return n;
            }

            int one = 1;
            int two = 2;
            int result = 0;
            for (int i = 0; i < n - 2; i++) {
                result = one + two;
                one = two;
                two = result;
            }

            return result;
        }
    }
}

File: leetcode_go/question/_1_100/_70/sol_70.go
// Package _70 爬楼梯
// @author goinhn
// @data 2022/8/19
package _70

// 动态规划
// time:O(n)
// space:O(1)
func climbStairs(n int) int {
	if n < 3 {
		return n
	}

	first := 1
	second := 2
	for i := 3; i <= n; i++ {
		first, second = second, first+second
	}

	return second
}
